src/target/riscv/riscv-013.c

```diff
@@ -1063,35 +1079,52 @@ static int is_vector_reg(uint32_t gdb_regno)
 		gdb_regno == GDB_REGNO_VLENB;
 }
 
-static int prep_for_register_access(struct target *target, uint64_t *mstatus,
-		int regno)
+static int prep_for_register_access(struct target *target,
+		riscv_reg_t *orig_mstatus, enum gdb_regno regno)
 {
-	if (is_fpu_reg(regno) || is_vector_reg(regno)) {
-		if (register_read_direct(target, mstatus, GDB_REGNO_MSTATUS) != ERROR_OK)
-			return ERROR_FAIL;
-		if (is_fpu_reg(regno) && (*mstatus & MSTATUS_FS) == 0) {
-			if (register_write_direct(target, GDB_REGNO_MSTATUS,
-						set_field(*mstatus, MSTATUS_FS, 1)) != ERROR_OK)
-				return ERROR_FAIL;
-		} else if (is_vector_reg(regno) && (*mstatus & MSTATUS_VS) == 0) {
-			if (register_write_direct(target, GDB_REGNO_MSTATUS,
-						set_field(*mstatus, MSTATUS_VS, 1)) != ERROR_OK)
-				return ERROR_FAIL;
-		}
-	} else {
-		*mstatus = 0;
-	}
+	if (!is_fpu_reg(regno) && !is_vector_reg(regno))
```

Comment:
Detail: For additional NULL safety, you can add an assert here:

```suggestion
	assert(orig_mstatus);

	if (!is_fpu_reg(regno) && !is_vector_reg(regno))
```

Reply:
Addressed.